src/setup_payload/ManualSetupPayloadGenerator.cpp

```diff
@@ -152,6 +152,9 @@ CHIP_ERROR ManualSetupPayloadGenerator::payloadDecimalStringRepresentation(Mutab
     int checkDigit = Verhoeff10::CharToVal(Verhoeff10::ComputeCheckChar(outBuffer.data()));
     ReturnErrorOnFailure(decimalStringWithPadding(outBuffer.SubSpan(offset, 2), static_cast<uint32_t>(checkDigit)));
 
+    // Reduce size of outBuffer by 1 to not include null terminator.
+    outBuffer.reduce_size(outBuffer.size() - 1);
```

Comment:
Again, this should be setting the span size to the actual amount of data written.  Probably `offset + 1` in this case, unless we increase `offset` after the check digit like we do in the other cases.